x/mint/types/minter_test.go

```diff
@@ -109,10 +110,22 @@ func TestCalculateBlockProvision(t *testing.T) {
 			// 80 billion utia (annual provisions) * 30 (seconds) / 31,556,952 (seconds per year) = 76052.97241635 which truncates to 76052 utia
 			want: sdk.NewCoin(DefaultBondDenom, sdk.NewInt(76052)),
 		},
+		{
+			name:             "want error when current time is before previous time",
+			annualProvisions: annualProvisions,
+			current:          current,
+			previous:         current.Add(blockInterval),
+			wantErr:          true,
+		},
 	}
 	for _, tc := range testCases {
 		minter.AnnualProvisions = tc.annualProvisions
-		got := minter.CalculateBlockProvision(tc.current, tc.previous)
+		got, err := minter.CalculateBlockProvision(tc.current, tc.previous)
+		if tc.wantErr {
+			assert.Error(t, err)
+			return
+		}
+		assert.NoError(t, err)
```

Comment:
[optional]
```suggestion
	assert.Equal(t, err!=nil, tc.wantErr)
```

Reply:
I think the order of arguments in this suggestion is incorrect because the second argument should be expected and the third argument should be actual.

Reply:
The code in this PR is consistent with other tests in this repo. I don't feel strongly about which approach is preferred but don't want to wait for another two approvals so if you feel strongly about the suggestion, please create a GH issue and we can tackle it repo-wide.

Reply:
My point was to suggest replacing the if statement with a one-line equality check, and you can, of course, switch the order of arguments.